src/plugins/intel_npu/src/plugin/npuw/weights_bank.hpp

```diff
@@ -26,18 +26,26 @@ class Bank {
         : m_core(core),
           m_alloc_device(alloc_device) {}
 
-    // Allocate a new tensor (if needed) on a specified device. LazyTensor needs to be registered and evaluated first
-    ov::Tensor get(const LazyTensor& tensor, const std::string& device);
-    // Register LazyTensor in a bank if it's not there
-    void registerLT(const LazyTensor& tensor, const std::string& device);
+    // Register LazyTensor in a bank if it's not there. Returns LazyTensor's unique id
+    int64_t registerLT(const LazyTensor& tensor, const std::string& device);
+
+    // Allocate and evaluate a registered tensor on a specified device (if needed) and return it from the bank
+    ov::Tensor get(int64_t uid, const std::string& device);
+
```

Comment:
As you just mentioned, now it neither evaluates nor allocates the tensor. Just gets it